File: frontend/src/types/facility.ts
export interface FacilityItem {
  id: number
  name: string
  type: 'COLD_STORAGE' | 'REFRIGERATED_VEHICLE' | string
  lng: number
  lat: number
  status: 'ONLINE' | 'OFFLINE' | 'ALARM' | string
  currentTemp?: number
  currentHumidity?: number
  capacity?: number
  usedCapacity?: number
  loadRate?: number
  address?: string
  plateNo?: string
  driverName?: string
  driverPhone?: string
  speed?: number
  updateTime?: string
}

export interface FacilityListData {
  coldStorages: FacilityItem[]
  vehicles: FacilityItem[]
  total: { storages: number; vehicles: number }
}

export interface SensorDevice {
  sensorId: number
  facilityId: number
  name: string
  type: string
  model?: string
  status: number
}

export interface FacilityDetail {
  facility: FacilityItem
  sensors: SensorDevice[]
}

export interface TrackPoint {
  lng: number
  lat: number
  speed?: number
  reportTime: string
}

File: backend/src/main/java/com/coldchain/modules/facility/entity/Facility.java
package com.coldchain.modules.facility.entity;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import lombok.Data;

import java.time.LocalDateTime;

/**
 * 冷库 / 冷藏车设施实体
 */
@Data
@TableName("facility")
public class Facility {

    @TableId(type = IdType.AUTO)
    private Long facilityId;
    private String name;
    /** COLD_STORAGE / REFRIGERATED_VEHICLE */
    private String type;
    private Double lng;
    private Double lat;
    /** ONLINE / OFFLINE / ALARM */
    private String status;
    private Double capacity;
    private Double usedCapacity;
    private String address;
    private Double currentTemp;
    private Double currentHumidity;
    private Double loadRate;
    private LocalDateTime updateTime;
    private LocalDateTime createTime;
}
